### PR TITLE
feat(chat): Make chat push content to the side in large view

### PR DESCRIPTION
**Pull request by muscat1**
Relates to https://github.com/jitsi/jitsi-meet/pull/5124

### CODE REVIEW

File: react/features/large-video/components/LargeVideo.web.js [hunk 1]
```diff
@@ -85,8 +90,11 @@ class LargeVideo extends Component<Props> {
  */
 function _mapStateToProps(state) {
     const testingConfig = state['features/base/config'].testing;
+    const { isOpen: shiftRight } = state['features/chat'];
```
Comment (saghul) [inline react/features/large-video/components/LargeVideo.web.js]: Can you build the class names in the render method instead? Pass some chatOpen flag on mapStateToProps

File: modules/UI/videolayout/LargeVideoManager.js [hunk 2]
```diff
@@ -323,7 +323,14 @@ export default class LargeVideoManager {
      * Update container size.
      */
     updateContainerSize() {
-        this.width = UIUtil.getAvailableVideoWidth();
+        let widthToUse = UIUtil.getAvailableVideoWidth();
+        const { isOpen } = APP.store.getState()['features/chat'];
+
+        if (isOpen) {
+            widthToUse -= 375;
```
Comment (saghul) [inline modules/UI/videolayout/LargeVideoManager.js]: can you put a comment about where this width comes from?

File: css/_videolayout_default.scss [hunk 3]
```diff
@@ -181,6 +181,12 @@
         visibility: hidden;
         z-index: $zindex2;
     }
+
+    &.shift-right {
+        #largeVideoWrapper {
+            margin-left: 375px;
```
Comment (saghul) [inline css/_videolayout_default.scss]: isn't there a variable we can use?

File: react/features/large-video/components/LargeVideo.web.js (hunk 1 above)
Comment (muscat1) [reply react/features/large-video/components/LargeVideo.web.js]: Done

File: modules/UI/videolayout/LargeVideoManager.js (hunk 2 above)
Comment (muscat1) [reply modules/UI/videolayout/LargeVideoManager.js]: Done

File: css/_videolayout_default.scss (hunk 3 above)
Comment (muscat1) [reply css/_videolayout_default.scss]: You are correct, made the change

File: react/features/chat/components/web/Chat.js [hunk 4]
```diff
@@ -73,6 +74,9 @@ class Chat extends AbstractChat<Props> {
             this._scrollMessageContainerToBottom(true);
         } else if (this.props._isOpen && !prevProps._isOpen) {
             this._scrollMessageContainerToBottom(false);
+            VideoLayout.onResize();
```
Comment (hristoterezov) [inline react/features/chat/components/web/Chat.js]: Don't we have an action for opening? Maybe we can hook up the resize thing there?

Comment (muscat1) [reply react/features/chat/components/web/Chat.js]: Moved the resizing logic in `chat/actions`

File: modules/UI/videolayout/VideoContainer.js [hunk 5]
```diff
@@ -426,14 +426,17 @@ export class VideoContainer extends LargeContainer {
         const { horizontalIndent, verticalIndent }
             = this.getVideoPosition(width, height, containerWidth, containerHeight);
 
+        const { isOpen } = APP.store.getState()['features/chat'];
+        const horizontalIndentLeft = isOpen ? 0 : horizontalIndent;
```
Comment (hristoterezov) [inline modules/UI/videolayout/VideoContainer.js]: `getVideoPosition` is centring the content. Why do you need to set `left` to `0`?

**Comment by hristoterezov** [comment]
When I tested the PR I see the following issues:
 - When I'm testing the PR, I see the large video just moved to left edge of the window.
 - There is animation for the large video which is executed even when you initially join the call. Is this a desired behaviour?

File: modules/UI/videolayout/VideoContainer.js (hunk 5 above)
Comment (muscat1) [reply modules/UI/videolayout/VideoContainer.js]: Because left is set to a negative value here and thus a small portion of it is hidden behind the chat

**Comment by muscat1** [comment]
Updated the PR, also moving the `Subject` component to the video container, in order for it to move alongside the video content. Please LMK if that's an issue.

**Comment by Echolon** [comment]
@muscat1 

May you have some time to review the change request?
That would be great!

**Comment by saghul** [comment]
@vp8x8 can you please review?

File: modules/UI/videolayout/LargeVideoManager.js [hunk 6]
```diff
@@ -323,7 +323,18 @@ export default class LargeVideoManager {
...
-        this.width = UIUtil.getAvailableVideoWidth();
+        let widthToUse = UIUtil.getAvailableVideoWidth();
+        const { isOpen } = APP.store.getState()['features/chat'];
+
+        if (isOpen) {
+            /**
+             * If chat state is open, we re-compute the container width
+             * by subtracting the default width of the chat.
+             */
+            widthToUse -= 375;
```
Comment (hristoterezov) [inline modules/UI/videolayout/LargeVideoManager.js:334]: Minor: you can define this as constant on the top of the file.

File: react/features/large-video/components/LargeVideo.web.js [hunk 7]
```diff
@@ -4,6 +4,8 @@ import React, { Component } from 'react';
 
 import { Watermarks } from '../../base/react';
 import { connect } from '../../base/redux';
+import { default as InviteMore } from '../../conference/components/web/InviteMore';
+import { default as Subject } from '../../conference/components/web/Subject';
```
Comment (hristoterezov) [inline react/features/large-video/components/LargeVideo.web.js:8]: Why not:
```
import { InviteMore,  Subject} from '../../conference';
```

File: react/features/toolbox/functions.web.js [hunk 8]
```diff
@@ -43,8 +44,10 @@ export function isToolboxVisible(state: Object) {
         visible
     } = state['features/toolbox'];
     const { audioSettingsVisible, videoSettingsVisible } = state['features/settings'];
+    const { isOpen } = state['features/chat'];
+    const isMobileChatOpen = isMobileBrowser() && isOpen;
 
-    return Boolean(!iAmSipGateway && (timeoutID || visible || alwaysVisible
+    return Boolean(!isMobileChatOpen && !iAmSipGateway && (timeoutID || visible || alwaysVisible
```
Comment (hristoterezov) [inline react/features/toolbox/functions.web.js:50]: Why do we check if the browser is mobile here?

Comment (muscat1) [reply react/features/toolbox/functions.web.js:50]: @saghul raised a concern regarding the use case of the conference in a mobile browser, and this is to ensure that it doesn't look botched in a mobile browser.

Comment (hristoterezov) [reply react/features/toolbox/functions.web.js:50]: So no toolbar in mobile browsers if you have the chat opened?

Comment (muscat1) [reply react/features/toolbox/functions.web.js:50]: This was proposed as a solution by @vp8x8 in order to not clutter the UI on the mobile browser

**Comment by sushantdanekar** [comment]
Is there any setting added for  this change ? like shiftchatwindowright = true or false. ?

**Comment by saghul** [comment]
No, it just shifts the contents to the right since the chat is shown on the left.

**Pull request merged**